File: RevShop_P2/backend/src/main/java/com/revshop/service/NotificationService.java
package com.revshop.service;

import com.revshop.model.Notification;
import java.util.List;

public interface NotificationService {

    void createNotification(Long userId, String message);

    List<Notification> getUserNotifications(Long userId);

    List<Notification> getUnreadNotifications(Long userId);

    void markAsRead(Long notificationId);
}

File: RevShop_P2/frontend/src/app/core/services/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Notification {
    id: number;
    userId: number;
    message: string;
    read: boolean;
    createdAt: string;
}

@Injectable({
    providedIn: 'root'
})
export class NotificationService {

    private baseUrl = 'http://localhost:8080/api/notifications';

    constructor(private http: HttpClient) { }

    getUserNotifications(email: string): Observable<Notification[]> {
        return this.http.get<Notification[]>(`${this.baseUrl}/${email}`);
    }

    getUnreadNotifications(email: string): Observable<Notification[]> {
        return this.http.get<Notification[]>(`${this.baseUrl}/${email}/unread`);
    }

    markAsRead(id: number): Observable<any> {
        return this.http.put(`${this.baseUrl}/${id}/read`, {}, { responseType: 'text' });
    }
}
